apollo-normalized-cache/src/test/java/com/apollographql/apollo/cache/normalized/internal/CacheKeyBuilderTest.kt

```diff
@@ -0,0 +1,369 @@
+package com.apollographql.apollo.cache.normalized.internal
+
+import com.apollographql.apollo.api.InputType
+import com.apollographql.apollo.api.Operation
+import com.apollographql.apollo.api.ResponseField
+import com.apollographql.apollo.api.ResponseField.Companion.forString
+import com.apollographql.apollo.api.ScalarType
+import com.apollographql.apollo.api.internal.InputFieldMarshaller
+import com.apollographql.apollo.api.internal.InputFieldWriter
+import com.apollographql.apollo.api.internal.UnmodifiableMapBuilder
+import com.google.common.truth.Truth
+import org.junit.Test
+import java.io.IOException
+import java.math.BigDecimal
+import java.util.HashMap
+
+class CacheKeyBuilderTest {
```

Comment:
Just curious why this test was converted to Kotlin, is it required by this PR?

Reply:
In Kotlin, this test can make use of `InputFieldWriter.writeList(String, (ListItemWriter) -> Unit)` (the version that takes a block as parameter) and therefore reproduce the problem.